app/code/core/Mage/Core/Model/Cache.php

```diff
@@ -116,94 +83,34 @@ public function __construct(array $options = [])
         $backend    = $this->_getBackendOptions($options);
         $frontend   = $this->_getFrontendOptions($options);
 
-        $this->_frontend = Zend_Cache::factory(
-            'Varien_Cache_Core',
-            $backend['type'],
-            $frontend,
-            $backend['options'],
-            true,
-            true,
-            true,
-        );
-
-        if (isset($options['request_processors'])) {
-            $this->_requestProcessors = $options['request_processors'];
-        }
-
-        if (isset($options['disallow_save'])) {
-            $this->_disallowSave = (bool) $options['disallow_save'];
-        }
+        $this->_frontend = match ($backend['type']) {
+            'redis' => new \Symfony\Component\Cache\Adapter\RedisTagAwareAdapter(
+                \Symfony\Component\Cache\Adapter\RedisTagAwareAdapter::createConnection($backend['options']['dsn']),
+                "maho-{$this->_idPrefix}",
+                $frontend['lifetime'],
+                new \Maho\CacheMarshaller(),
```

Comment:
What do you think about putting our marshaller class at `app/code/core/Mage/Core/Model/Cache/Marshaller.php` or similar, and replacing the above line with `Mage::getModel('core/cache_marshaller')` ?

I think there could be some situations where a module wants to overwrite this, and it would be less great to require the developer to override `Mage_Core_Model_Cache`'s constructor.

Reply:
I don't super too much love it but I think it's worth doing it :-)))) just pushed it